server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryResponse.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportResponse;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: response from extensions for name writeable registry entries
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryResponse extends TransportResponse {
+
+    private final Map<String, Class> registry;
+
+    /**
+     * @param registry map of writeable names and their associated category class
+     */
+    public NamedWriteableRegistryResponse(Map<String, Class> registry) {
+        this.registry = new HashMap<>(registry);
+    }
+
+    /**
+     * @param in StreamInput from which map entries of writeable names and their associated category classes are read from
+     * @throws IllegalArgumentException if the fully qualified class name is invalid and the class object cannot be generated at runtime
+     */
+    public NamedWriteableRegistryResponse(StreamInput in) throws IOException {
+        super(in);
+        // stream output for registry map begins with a variable integer that tells us the number of entries being sent across the wire
```

Comment:
```suggestion
        // Stream output for registry map begins with a variable integer that tells us the number of entries being sent across the wire
```
Here and everywhere :slightly_smiling_face: